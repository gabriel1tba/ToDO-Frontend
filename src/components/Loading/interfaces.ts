export interface ILoading {
  isLoading?: boolean;
  onTop?: boolean;
  size: number;
  color?: 'white' | 'main';
}
