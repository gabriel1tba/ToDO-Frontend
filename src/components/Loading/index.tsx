import ReactDOM from 'react-dom';

import * as S from './styles';

import { ILoading } from './interfaces';

const Loading = ({
  isLoading = true,
  onTop = false,
  color = 'main',
  size,
}: ILoading) => {
  const portalRoot = document.querySelector('#portal-loader-root') as Element;

  if (!isLoading) return null;

  return onTop ? (
    ReactDOM.createPortal(
      <S.Overlay>
        <S.Loading size={size} color={color} />
      </S.Overlay>,
      portalRoot
    )
  ) : (
    <S.Loading size={size} color={color} />
  );
};

export default Loading;
