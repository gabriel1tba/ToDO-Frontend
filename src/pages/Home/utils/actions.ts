import { TTodo } from 'services/TodoService/interfaces';

export enum Types {
  GET_TODOS = 'GET_TODOS',
  CREATE_TODO = 'CREATE_TODO',
  UPDATE_TODO = 'UPDATE_TODO',
  DELETE_TODO = 'DELETE_TODO',
}

interface GetTodos {
  type: Types.GET_TODOS;
  payload: TTodo[];
}

interface CreateTodo {
  type: Types.CREATE_TODO;
  payload: TTodo;
}

interface UpdateTodo {
  type: Types.UPDATE_TODO;
  payload: TTodo;
}

interface DeleteTodo {
  type: Types.DELETE_TODO;
  payload: string;
}

export type Action = GetTodos | CreateTodo | UpdateTodo | DeleteTodo;

export const getTodosAction = (todos: TTodo[]): GetTodos => ({
  type: Types.GET_TODOS,
  payload: todos,
});

export const createTodoAction = (todo: TTodo): CreateTodo => ({
  type: Types.CREATE_TODO,
  payload: todo,
});

export const updateTodoAction = (todo: TTodo): UpdateTodo => ({
  type: Types.UPDATE_TODO,
  payload: todo,
});

export const deleteTodoAction = (id: string): DeleteTodo => ({
  type: Types.DELETE_TODO,
  payload: id,
});
