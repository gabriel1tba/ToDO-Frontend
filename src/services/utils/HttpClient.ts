import axios from 'axios';

const HttpClient = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
});

export default HttpClient;
