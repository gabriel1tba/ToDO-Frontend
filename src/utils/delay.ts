const delay = (ms = 1000): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));

export default delay;
