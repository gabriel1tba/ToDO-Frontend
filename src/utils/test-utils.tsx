import { render, RenderOptions } from '@testing-library/react';
import { TodoProvider } from 'context/todos';

interface IWrapperProvider {
  children: React.ReactNode;
}

const WrapperProvider = ({ children }: IWrapperProvider) => (
  <TodoProvider>{children}</TodoProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(<WrapperProvider>{ui}</WrapperProvider>, { ...options });

export * from '@testing-library/react';
export { customRender as render };
