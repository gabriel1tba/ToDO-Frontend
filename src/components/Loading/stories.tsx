import { Story, Meta } from '@storybook/react';

import Loading from '.';
import { ILoading } from './interfaces';

export default {
  title: 'Loading',
  component: Loading,
  argTypes: {
    isLoading: {
      type: 'boolean',
    },
  },
} as Meta;

export const Default: Story<ILoading> = (args) => <Loading {...args} />;

Default.args = {
  isLoading: true,
};
