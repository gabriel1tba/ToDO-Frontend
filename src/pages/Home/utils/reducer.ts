import { TTodo } from 'services/TodoService/interfaces';

import { Types, Action } from './actions';

const todosReducer = (state: TTodo[], action: Action): TTodo[] => {
  switch (action.type) {
    case Types.GET_TODOS: {
      return action.payload;
    }

    case Types.CREATE_TODO: {
      return [...state, action.payload];
    }

    case Types.UPDATE_TODO: {
      return state.map((todo) => {
        if (action.payload.id === todo.id) {
          todo = { ...action.payload, created_at: todo.created_at };
        }
        return todo;
      });
    }

    case Types.DELETE_TODO: {
      return state.filter((todo) => {
        return todo.id !== action.payload;
      });
    }

    default:
      return state;
  }
};

export default todosReducer;
