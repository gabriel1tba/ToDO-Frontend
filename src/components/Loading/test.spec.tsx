import { render } from 'utils/test-utils';

import Loading from '.';

describe('<Loading />', () => {
  it('should render the Loading correctly', () => {
    const { container } = render(<Loading />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
