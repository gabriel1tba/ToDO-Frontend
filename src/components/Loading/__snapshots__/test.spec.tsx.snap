// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the Loading correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100vh;
}

.c0 .lds-roller {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}

.c0 .lds-roller div {
  -webkit-animation: lds-roller 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: lds-roller 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  -webkit-transform-origin: 40px 40px;
  -ms-transform-origin: 40px 40px;
  transform-origin: 40px 40px;
}

.c0 .lds-roller div:after {
  content: ' ';
  display: block;
  position: absolute;
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: #ff9000;
  margin: -4px 0 0 -4px;
}

.c0 .lds-roller div:nth-child(1) {
  -webkit-animation-delay: -0.036s;
  animation-delay: -0.036s;
}

.c0 .lds-roller div:nth-child(1):after {
  top: 63px;
  left: 63px;
}

.c0 .lds-roller div:nth-child(2) {
  -webkit-animation-delay: -0.072s;
  animation-delay: -0.072s;
}

.c0 .lds-roller div:nth-child(2):after {
  top: 68px;
  left: 56px;
}

.c0 .lds-roller div:nth-child(3) {
  -webkit-animation-delay: -0.108s;
  animation-delay: -0.108s;
}

.c0 .lds-roller div:nth-child(3):after {
  top: 71px;
  left: 48px;
}

.c0 .lds-roller div:nth-child(4) {
  -webkit-animation-delay: -0.144s;
  animation-delay: -0.144s;
}

.c0 .lds-roller div:nth-child(4):after {
  top: 72px;
  left: 40px;
}

.c0 .lds-roller div:nth-child(5) {
  -webkit-animation-delay: -0.18s;
  animation-delay: -0.18s;
}

.c0 .lds-roller div:nth-child(5):after {
  top: 71px;
  left: 32px;
}

.c0 .lds-roller div:nth-child(6) {
  -webkit-animation-delay: -0.216s;
  animation-delay: -0.216s;
}

.c0 .lds-roller div:nth-child(6):after {
  top: 68px;
  left: 24px;
}

.c0 .lds-roller div:nth-child(7) {
  -webkit-animation-delay: -0.252s;
  animation-delay: -0.252s;
}

.c0 .lds-roller div:nth-child(7):after {
  top: 63px;
  left: 17px;
}

.c0 .lds-roller div:nth-child(8) {
  -webkit-animation-delay: -0.288s;
  animation-delay: -0.288s;
}

.c0 .lds-roller div:nth-child(8):after {
  top: 56px;
  left: 12px;
}

<div
  class="c0"
>
  <div
    class="lds-roller"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
